### PR TITLE
Implement VBNRunner and basic constraint negation

### PR DESCRIPTION
**Pull request by brendansaw**

### CODE REVIEW

File: src/main/java/vbn/state/State.java [hunk 1]
```diff
@@ -110,6 +110,10 @@ public void pushConstraint(UnaryOperand operand, String symbol) {
         pushConstraint(constraint);
     }
 
+    public void pushConstraints(Stack<Constraint> constraints) {
```
Comment (chrishappy) [inline src/main/java/vbn/state/State.java]: This looks a little sketchy. Could you double-check?

File: src/main/java/vbn/state/constraints/BinaryConstraint.java [hunk 2]
```diff
@@ -18,12 +18,14 @@ public BinaryConstraint(@NonNull Symbol left, @NonNull BinaryOperand op, @NonNul
...
         this.op = op;
         this.right = right;
+        this.negated = false;
     }
     public BinaryConstraint(@NonNull Symbol left, @NonNull BinaryOperand op, @NonNull Symbol right, Symbol assigned) {
         this.assigned = assigned;
         this.left = left;
         this.op = op;
         this.right = right;
+        this.negated = false;
```
Comment (chrishappy) [inline src/main/java/vbn/state/constraints/BinaryConstraint.java:28]:
```suggestion
```

If you could remove this too...

File: src/main/java/vbn/state/constraints/BinaryConstraint.java [hunk 3]
```diff
@@ -18,12 +18,14 @@ public BinaryConstraint(@NonNull Symbol left, @NonNull BinaryOperand op, @NonNul
...
```
Comment (chrishappy) [inline src/main/java/vbn/state/constraints/BinaryConstraint.java:21]:
```suggestion
```

If you could remove this...

File: src/main/java/vbn/state/constraints/Constraint.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package vbn.state.constraints;
 
 public abstract class Constraint {
+    public boolean negated;
```
Comment (chrishappy) [inline src/main/java/vbn/state/constraints/Constraint.java:4]:
```suggestion
```
and this...

File: src/main/java/vbn/state/constraints/UnaryConstraint.java [hunk 5]
```diff
@@ -16,11 +16,13 @@ public UnaryConstraint(@NonNull UnaryOperand op, @NonNull Symbol symbol) {
         this.assigned = null;
         this.symbol = symbol;
         this.op = op;
+        this.negated = false;
```
Comment (chrishappy) [inline src/main/java/vbn/state/constraints/UnaryConstraint.java:19]:
```suggestion
```

File: src/main/java/vbn/state/constraints/UnaryConstraint.java [hunk 6]
```diff
@@ -16,11 +16,13 @@ public UnaryConstraint(@NonNull UnaryOperand op, @NonNull Symbol symbol) {
...
     }
     public UnaryConstraint(@NonNull UnaryOperand op, @NonNull Symbol symbol, Symbol assigned) {
         this.assigned = assigned;
         this.symbol = symbol;
         this.op = op;
+        this.negated = false;
```
Comment (chrishappy) [inline src/main/java/vbn/state/constraints/UnaryConstraint.java:25]:
```suggestion
```

**Review by chrishappy — CHANGES_REQUESTED** [review]

File: src/main/java/vbn/state/constraints/BinaryConstraint.java (hunk 3 above)
Comment (brendansaw) [reply src/main/java/vbn/state/constraints/BinaryConstraint.java]: Fixed in [c20af33](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/c20af330db3eb5857b775153c9499980e5da16a4) and [cefec5c](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/cefec5c6ca5e3350d7c07a128c30b406a65d884a)

File: src/main/java/vbn/state/constraints/Constraint.java (hunk 4 above)
Comment (brendansaw) [reply src/main/java/vbn/state/constraints/Constraint.java]: Fixed in [c20af33](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/c20af330db3eb5857b775153c9499980e5da16a4) and [cefec5c](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/cefec5c6ca5e3350d7c07a128c30b406a65d884a)

File: src/main/java/vbn/state/constraints/UnaryConstraint.java (hunk 5 above)
Comment (brendansaw) [reply src/main/java/vbn/state/constraints/UnaryConstraint.java]: Fixed in [c20af33](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/c20af330db3eb5857b775153c9499980e5da16a4) and [cefec5c](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/cefec5c6ca5e3350d7c07a128c30b406a65d884a)

File: src/main/java/vbn/state/constraints/UnaryConstraint.java (hunk 6 above)
Comment (brendansaw) [reply src/main/java/vbn/state/constraints/UnaryConstraint.java]: Fixed in [c20af33](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/c20af330db3eb5857b775153c9499980e5da16a4) and [cefec5c](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/cefec5c6ca5e3350d7c07a128c30b406a65d884a)

File: src/main/java/vbn/state/constraints/BinaryConstraint.java (hunk 2 above)
Comment (brendansaw) [reply src/main/java/vbn/state/constraints/BinaryConstraint.java]: Fixed in [c20af33](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/c20af330db3eb5857b775153c9499980e5da16a4) and [cefec5c](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/cefec5c6ca5e3350d7c07a128c30b406a65d884a)

File: src/main/java/vbn/state/State.java (hunk 1 above)
Comment (brendansaw) [reply src/main/java/vbn/state/State.java]: Fixed in [2f8565f](https://github.com/CMPT-479/CMPT-479-Project/pull/32/commits/2f8565f4f3df431c55656d09b9ba55c80fefca89)

**Review by chrishappy — APPROVED** [review]
Thanks for doing this!

**Comment by chrishappy** [comment]
Tests fail due to not all variables passing.

**Pull request merged**